janus_server/src/datastore.rs

```diff
@@ -128,6 +128,15 @@ impl<C: Clock> Datastore<C> {
         tx.tx.commit().await?;
         Ok(rslt)
     }
+
+    /// Attempt to fetch a connection from the connection pool, to check if
+    /// the database is accessible. This will either create a new database
+    /// connection or recycle an existing one, and then return the connection
+    /// back to the pool.
+    pub async fn check_connection_pool(&self) -> Result<(), Error> {
```

Comment:
#322 adds retries to getting connections from the pool, so I think we'll want to refactor this so that both this and `run_tx_once` use the same code to get a connection... but I can take care of that by rebasing #322 after we merge this change.